### PR TITLE
fix: index error fixed for item_alternative query

### PR DESCRIPTION
**Pull request by vishdha**
[TASK ID](https://app.asana.com/0/1192407894480744/1200314326901032)

Peek 2021-05-19 18-51

### CODE REVIEW

File: erpnext/stock/doctype/item_alternative/item_alternative.py [hunk 1]
```diff
@@ -43,13 +43,26 @@ def validate_duplicate(self):
 			frappe.throw(_("Already record exists for the item {0}".format(self.item_code)))
 
 def get_alternative_items(doctype, txt, searchfield, start, page_len, filters):
-	return frappe.db.sql(""" (select alternative_item_code from `tabItem Alternative`
-			where item_code = %(item_code)s and alternative_item_code like %(txt)s)
+	return frappe.db.sql("""
+		(select
+			alternative_item_code, alternative_item_name
```
Comment (hrwX) [inline erpnext/stock/doctype/item_alternative/item_alternative.py]: - These fields should be dynamically generated based on title field
- check queries.py for reference

**Review by hrwX — DISMISSED** [review]

File: erpnext/stock/doctype/item_alternative/item_alternative.json [hunk 2]
```diff
@@ -1,12 +1,12 @@
 {
+ "autoname": "field:alternative_item_code",
```
Comment (hrwX) [inline erpnext/stock/doctype/item_alternative/item_alternative.json:2]: Can we not have more than one alternative items linked with an item ?

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/item_alternative/item_alternative.json (hunk 2 above)
Comment (vishdha) [reply erpnext/stock/doctype/item_alternative/item_alternative.json:2]: nope

File: erpnext/stock/doctype/item_alternative/item_alternative.py [hunk 3]
```diff
@@ -43,13 +44,21 @@ def validate_duplicate(self):
...
-			"item_code": filters.get('item_code'),
-			"txt": '%' + txt + '%'
+	fields = get_fields("Item Alternative", ["alternative_item_code", "item_code", "alternative_item_name"])
+	return frappe.db.sql("""
+		(select
+			{fields}
+		from
+			`tabItem Alternative`
+		where
+			item_code = %(item_code)s and alternative_item_code like %(txt)s)
```
Comment (hrwX) [inline erpnext/stock/doctype/item_alternative/item_alternative.py]:
```suggestion
		select
			{fields}
		from
			`tabItem Alternative`
		where
			item_code = %(item_code)s and alternative_item_code like %(txt)s
```

**Review by hrwX — DISMISSED** [review]

**Review by hrwX — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**